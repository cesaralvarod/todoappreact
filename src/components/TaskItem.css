li.task-item {
  display: flex;
  align-items: flex-start;
  user-select: none;
  position: relative;
  margin-bottom: 1rem;
}

li.task-item:last-child {
  margin-bottom: 0;
}

label.task-content {
  position: relative;
  cursor: pointer;
}

input[type="checkbox"] {
  display: none;
}

span.task-checkbox {
  width: 1.5rem;
  height: 1.5rem;
  content: "";
  position: absolute;
  top: 1px;
  left: 0;
  background: transparent;
  border: 1px solid #aaa;
  border-radius: 4px;
  color: #fff;
  text-align: center;
  font-size: 1.2rem;
  overflow: hidden;
  transition: all 0.3s ease-out;
}

span.task-checkbox::before {
  content: "";
  top: 50%;
  left: 50%;
  transform: translateX(-50%);
  transition: all 0.3s ease-out;
}

input[type="checkbox"]:checked + label.task-content > span.task-checkbox {
  content: "✓";
  border: 1px solid #08c42a;
  background: #08c42a;
}

input[type="checkbox"]:checked
  + label.task-content
  > span.task-checkbox::before {
  content: "✓";
}

label.task-content p {
  padding: 0 2.5rem;
}

input[type="checkbox"]:checked + label.task-content > p {
  text-decoration: line-through;
}

button.task-times {
  background: none;
  border: none;
  font-size: 3rem;
  padding: 0;
  color: #c23a3a;
  cursor: pointer;
  position: absolute;
  top: -11px;
  right: 10px;
  transition: all 0.3s ease-out;
}

button.task-times:hover {
  color: #fa1111;
}
