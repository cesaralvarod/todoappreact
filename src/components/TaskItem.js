import React from "react";
import PropTypes from "prop-types";

// Styles

import "./TaskItem.css";

const TaskItem = ({ dataTask }) => {
  const { id, content, isComplete } = dataTask;

  return (
    <li className="task-item">
      <input type="checkbox" id={id} defaultChecked={isComplete} />
      <label htmlFor={id} className="task-content">
        <span className="task-checkbox"></span>
        <p>{content}</p>
      </label>
      <button className="task-times">&#215;</button>
    </li>
  );
};

TaskItem.propTypes = {
  dataTask: PropTypes.exact({
    id: PropTypes.number.isRequired,
    content: PropTypes.string.isRequired,
    isComplete: PropTypes.bool.isRequired,
  }),
};

export default TaskItem;
