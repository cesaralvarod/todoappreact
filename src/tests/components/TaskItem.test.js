import React from "react";
import { shallow } from "enzyme";
import TaskItem from "../../components/TaskItem";

describe("Testing in component TaskItem", () => {
  test("Component TaskItem should show correctly", () => {
    const dataParams = {
      dataTask: { id: 123, content: "lorem ipsum", done: false },
      handleToggle: jest.fn(),
      handleDelete: jest.fn(),
    };

    const wrapper = shallow(<TaskItem {...dataParams} />);

    expect(wrapper).toMatchSnapshot();
  });
});
