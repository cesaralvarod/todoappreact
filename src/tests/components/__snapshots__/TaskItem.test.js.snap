// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component TaskItem Component TaskItem should show correctly 1`] = `
<li
  className="task-item"
>
  <input
    defaultChecked={false}
    id={123}
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="task-content"
    htmlFor={123}
  >
    <span
      className="task-checkbox"
    />
    <p>
      lorem ipsum
    </p>
  </label>
  <button
    className="task-times"
    onClick={[Function]}
  >
    Ã—
  </button>
</li>
`;
